ocs_ci/deployment/deployment.py

```diff
@@ -247,3 +241,24 @@ def add_node(self):
         Implement platform specif add_node in child class
         """
         raise NotImplementedError("add node functionality node implemented")
+
+    def patch_default_sc_to_non_default(self):
+        """
+        Patch storage class which comes as default with installation to non-default
+        """
+        sc_to_patch = None
+        platform = config.ENV_DATA.get('platform')
+        if platform in constants.AWS_PLATFORM:
+            sc_to_patch = constants.DEFAULT_SC_AWS
```

Comment:
If we will define property for default_sc or sc_to_patch  in child class for specific deployment platform we can directly access it here without need of any condition and once new platform added this code doesn't have to be changed.

We will just require the new one deployment class for specific platform will implement this property.  WDYT? 

On line https://github.com/red-hat-storage/ocs-ci/blob/56b376da01327b01e96633ea011ffe653fd09a1a/ocs_ci/deployment/deployment.py#L31 . You can define 

self.default_sc or self.sc_to_patch as None and this have to be defined in child class here: https://github.com/red-hat-storage/ocs-ci/blob/56b376da01327b01e96633ea011ffe653fd09a1a/ocs_ci/deployment/aws.py#L33

Reply:
I agree with the above mentioned approach. Currently we don't have vSphere classes to assign default_sc. purpose of this patch is to use ocs-ci for installing ocs4 on vsphere ( ocp4 installation is manual and then use ocs-ci for the rest ). without this patch, installing ocs through rook will fail since gp2 doesn't exist in vsphere env.

I will address this in next PR

Reply:
I think that you will fail even sooner than the code will get here. Cause we have factory which returns you proper deployment class depends on the platform and deployment type (upi, ipi) as you can see here:

https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/factory.py#L40

As we don't have vsphere platform defined you will end up with KeyError exception ;)

So we need to define some basic deployment class anyway or how does it work for you?

Reply:
Added vSphere dummy classes for this PR. Actual implementation will take care in next PR.